src/commands/notify.yml

```diff
@@ -48,6 +48,21 @@ parameters:
         Disable to catch initial setup errors. Re-enable to prevent Slack errors from affecting your pipeline.
       type: boolean
       default: true
+  icon:
```

Comment:
Followed up in my initial comment. The icon seen in our provided templates are a part of the _template_, we do not inject icons into messages. For custom icons/messages I would currently recommend using a custom template.

What would be expected if a user set an icon parameter as well as a custom template?

Reply:
Trying to avoid custom templates since they need to be defined inline, which makes things less readable.  Maybe I can make the template assigned in an ENV var, so that it's abstracted to a file that I `cat` during a `run` step?

I also don't see anything that suggests a template lets me change the equiv of `SLACK_PARAM_ICON`, unless you mean the `accessory` example?

Reply:
You absolutely can 👍 , in fact, I don't believe we have ever really advertised this exact fact but placing templates in a context may be most convenient.
https://circleci.com/docs/2.0/contexts/

Here is how you can utilize templates from environment variables: https://github.com/CircleCI-Public/slack-orb/wiki/Dynamic-Templates